x/incentive/keeper/keeper.go

```diff
@@ -981,3 +981,89 @@ func (k Keeper) GetAllClaims(ctx sdk.Context) types.Claims {
 
 	return cs
 }
+
+// GetRewardAccrualTime fetches the last time rewards were accrued for the
+// specified ClaimType and subKey.
+func (k Keeper) GetRewardAccrualTime(
+	ctx sdk.Context,
+	claimType types.ClaimType,
+	subKey string,
+) (time.Time, bool) {
+	store := prefix.NewStore(ctx.KVStore(k.key), types.GetPreviousRewardAccrualTimeKeyPrefix(claimType))
+	b := store.Get([]byte(subKey))
```

Comment:
nit: I'd maybe use a function to convert the subKey into a byte key. Then there's just one place that defines the conversion.